modules/xfem/include/userobjects/MeshCut3DUserObject.h

```diff
@@ -70,27 +107,40 @@ class MeshCut3DUserObject : public GeometricCutUserObject
   /// Number of steps to grow the mesh
   unsigned int _n_step_growth;
 
-  /// variables to help control the work flow
+  /// Variables to help control the work flow
   bool _stop;
   bool _grow;
 
   /// Boundary nodes of the cutter mesh
   std::vector<dof_id_type> _boundary;
 
+  /// Active boundary nodes where growth is allowed
+  std::vector<std::vector<dof_id_type>> _active_boundary;
+
+  /// Inactive boundary
+  std::vector<unsigned int> _inactive_boundary_pos;
+
+  /// Front nodes that are grown from the crack front definition defined in the input
+  /// therefore, they are (1) in the same order as defined in the input and (2) the number of nodes does not change
+  std::vector<dof_id_type> _tracked_crack_front_points;
+
+  bool _cfd;
+
   /// Edges at the boundary
   std::set<Xfem::CutEdge> _boundary_edges;
 
   /// A map of boundary nodes and their neighbors
   std::map<dof_id_type, std::vector<dof_id_type>> _boundary_map;
 
-  /// Active boundary nodes where growth is allowed
-  std::vector<std::vector<dof_id_type>> _active_boundary;
-
   /// Growth direction for active boundaries
   std::vector<std::vector<Point>> _active_direction;
 
-  /// Inactive boundary
-  std::vector<unsigned int> _inactive_boundary_pos;
+  /// Growth size for the active boundary in a subcritical simulation
+  std::vector<Real> _growth_size;
+
+  /// Fatigue life
+  std::vector<unsigned long int> _dn;
+  std::vector<unsigned long int> _n;
```

Comment:
Do we have to put fatigue related parameters to this cut UO? I feel that the MeshCut3DUO should be general, and not targeted on only specific crack growth law. `ParisLaw` seems to be a good place for them.

Reply:
_dn and _n are not specific to growth law.  As long as the crack has subcritical growth, _n will be useful.  I feel it may be good to leave them here so the code can manage, e.g. the output of these numbers, disregard of the specific fatigue/SCC criteria.